### ISSUE TITLE
Adding to library leads to duplicate items weird animation

### ISSUE THREAD

**Issue by ad1992** [excalidraw/excalidraw#3317]
![Excalidraw (3)](https://user-images.githubusercontent.com/11256141/112172851-10469980-8c1b-11eb-95fd-c44ecd3d2147.gif)

Steps to reproduce
1. Create a shape
2. Add two library
3. Issue replicated

It leads to duplicated items out of which first one is correct, second one is the duplicate one. The dimensions of the duplicated one keeps changing hence it looks like animated. The action also is incorrect (it shows add icon instead of removal).

**Comment by h7y** [comment]
I think this shows up because it assumes the shape is not added yet and the animation is to prompt the user to add it. But it is confusing.

**Comment by ad1992** [comment]
The main issue is item gets added twice which shouldn't happen, once that is fixed the other issues would probably get automatically fixed

**Comment by h7y** [comment]
The item doesn't get added twice. If you hover over the animating shape, plus button shows up showing that it's not added yet unless the user clicks. Exporting without adding the animating shape doesn't result in duplicate item as well. 

I think it is done that way just to prompt the user that the shape he/she selected can be added to the library. The way it is done is confusing.

**Comment by ad1992** [comment]
yeah I get what you are saying, but somewhere some mismatch is happening due to which it tries to show the selected item can still be added to the library where as it was already added in reality and thats what needs to be fixed.

**Comment by dwelle** [comment]
As @h7y, there's no bug. We simply haven't yet implemented any duplication checks.

But I'm not sure we should do any de-duplication prevention in this case because it would lead to more confusion than not — people would wonder why they can add items in some cases and in some not. Maybe if the directly previous item is identical so it's obvious, but even then I'd be hesitant to do so — unless we made it obvious why you can't add it (e.g. gray it out with a hover tooltip explaining it).

There's also the question of how to figure out if an item is duplicate. We'd have to compare every property of every element to every element of every library item.

**Comment by ad1992** [comment]
Yeah I realised after checking the code that right now we are allowing duplicated items but I got confused by the behaviour of allowing duplicated items, not sure if we should allow duplicated items because that might confuse user as well.

> There's also the question of how to figure out if an item is duplicate. We'd have to compare every property of every element to every element of every library item.

Yeah thats the naive way but we might have to optimize this if we decide to not allow duplicate items.
